MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/AccountRetrievalService.java

```diff
@@ -0,0 +1,165 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.model.usermodel.UserResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+/**
+ * Background service which retrieves Mapbox Account information
+ */
+
+public class AccountRetrievalService extends IntentService {
+
+  private static final String BASE_URL = "https://api.mapbox.com/api/";
+  private static final String ACCESS_TOKEN_URL = "https://api.mapbox.com/oauth/access_token";
+
+  //  TODO: Fill in CLIENT_SECRET
+  private static final String CLIENT_SECRET = "";
+
+  private static String CLIENT_ID;
+  private static String REDIRECT_URI;
+  private String username;
+
+  public AccountRetrievalService() {
+    super("AccountRetrievalService");
+  }
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+  }
+
+  @Override
+  protected void onHandleIntent(@Nullable Intent intent) {
+    if (intent != null) {
+      String authCode = intent.getStringExtra("AUTHCODE");
+      CLIENT_ID = intent.getStringExtra("CLIENT_ID");
+      REDIRECT_URI = intent.getStringExtra("REDIRECT_URI");
+      getAccessToken(authCode);
+    } else {
+      Log.d("AccountRetrievalService", "onHandleIntent: intent == null");
+    }
+  }
+
+  private void getAccessToken(String code) {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+      .addHeader("User-Agent", "Android Dev Preview")
+      .addHeader("Content-Type", "application/x-www-form-urlencoded")
+      .url(ACCESS_TOKEN_URL)
+      .post(RequestBody.create(MediaType.parse("application/x-www-form-urlencoded"),
+        "grant_type=authorization_code&client_id=" + CLIENT_ID + "&client_secret=" + CLIENT_SECRET
+          + "&redirect_uri=" + REDIRECT_URI + "&code=" + code))
+      .build();
+
+    client.newCall(request).enqueue(new okhttp3.Callback() {
```

Comment:
The oAuth flow example I found online, used `okhttp3`. It's working fine as is, so I'm very inclined to just keep using `okhttp3` for this particular step, rather than spending time to figure out how to switch it to Retrofit2. That said, I looked into switching and I thought I was close.

In `MapboxAccountRetrofitService`, I have:
```java
@FormUrlEncoded
  @POST("grant_type=authorization_code&client_id={clientId}&client_secret={clientSecret}"
    + "&redirect_uri={redirectUri}&code={code}")
  Call<Object> getAccessToken(@Header("User-Agent") String userAgentHeader,
                              @Field("clientId") String clientId,
                              @Field("clientSecret") String clientSecret,
                              @Field("redirectUri") String redirectUri,
                              @Field("code") String code);
```
☝️ `Call<Object>` might not be the correct call class...

In `AccountRetrievalService`, I have the code below the screenshot. When I sign in and run everything, `Log.d("RetrievalService", "onResponse: response received!");` _does_ appear in the log 😕 

However 👇 
<img width="1215" alt="screen shot 2017-05-19 at 3 08 24 pm" src="https://cloud.githubusercontent.com/assets/4394910/26268637/1853a6b4-3ca5-11e7-818d-460c201ed90c.png">

```java
     Retrofit retrofit = new Retrofit.Builder()
      .baseUrl(ACCESS_TOKEN_URL)
      .addConverterFactory(GsonConverterFactory.create())
      .build();
    MapboxAccountRetrofitService service = retrofit.create(MapboxAccountRetrofitService.class);
    retrofit2.Call<Object> requestRequest = service.getAccessToken("Android Dev Preview",
      clientId, CLIENT_SECRET, redirectUri, code);

    requestRequest.enqueue(new retrofit2.Callback<Object>() {
      @Override
      public void onResponse(retrofit2.Call<Object> call, retrofit2.Response<Object> response) {
        Log.d("RetrievalService", "onResponse: response received!");
        String json = response.toString();
        String jsonString = response.raw().body().toString();
        Log.d("RetrievalService", "onResponse: json = " + json);
        Log.d("RetrievalService", "onResponse: jsonString = " + jsonString);
        JSONObject data = null;
        try {
          data = new JSONObject(json);
          String accessToken = data.optString("access_token");
          try {
            getUsernameFromJwt(accessToken);
          } catch (Exception exception) {
            exception.printStackTrace();
          }
          getUserInfo(username, accessToken);
        } catch (JSONException exception) {
          exception.printStackTrace();
        }
      }

      @Override
      public void onFailure(retrofit2.Call<Object> call, Throwable throwable) {
        throwable.printStackTrace();
      }
    });
```

@Guardiola31337 , fine to keep using `okhttp3` or is it important to use Retrofit right now for this one step? All of the code above hasn't been actually committed to this branch, so you'll have to copy/paste if you want to test it out.